lib/matplotlib/collections.py

```diff
@@ -899,6 +900,125 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_elements(self, prop="colors", num="auto",
+                     fmt=None, func=lambda x: x, **kwargs):
+        """
+        Creates legend handles and labels for a PathCollection. This is useful
+        for obtaining a legend for a :meth:`~.Axes.scatter` plot. E.g.::
+
+            scatter = plt.scatter([1,2,3], [4,5,6], c=[7,2,3])
+            plt.legend(*scatter.legend_elements())
+
+        Also see the :ref:`automatedlegendcreation` example.
+
+        Parameters
+        ----------
+        prop : string, optional, default *"colors"*
+            Can be *"colors"* or *"sizes"*. In case of *"colors"*, the legend
+            handles will show the different colors of the collection. In case
+            of "sizes", the legend will show the different sizes.
+        num : int, None, "auto" (default), or `~.ticker.Locator`, optional
```

Comment:
Although the ability to use a `Locator` is good, and provides the ability to put in a list via a `FixedLocator`, wouldn't one of the most common--maybe *the* most common--use cases be an explicit list?  For example, sizes represent populations of cities, so the legend shows sizes for `[10000, 100000, 1000000, 10000000]`.   If this would be a common use case, then it might be worth supporting it directly by accepting a sequence.  To make all this more like `contour` with its 'levels', a separate kwarg could be used to specify a `Locator` for the rare cases when it is needed.

Reply:
I would still use the same argument `num` though, because a list of values is an alternative to those other options. Else one would need to go through checking what happens if both are provided etc. 

So what is a canonical way of checking if something is a list, tuple or array of numbers? (`isinstance(x, collections.abc.Iterable)` only does half the job; is there something in `cbook` that I overlooked?)

Reply:
`np.iterable()` should suffice, I think.